### DISCUSSION TITLE
Error while building windows installer or application with runtime embedded

### DISCUSSION THREAD

**Discussion by balaedit** [airsdk/Adobe-Runtime-Support#2971] [General]
Error building the bundle for application with runtime embedded, find the images given below :

In Adobe Animate - 
![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/37571867/031172ce-5161-4c0d-b08c-e0a64e9a8589)

In Air SDK Manager - 
![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/37571867/713a6064-9753-41cc-bdbe-4cdb93f01bd6)

adt.cfg setting are default.

**Comment by balaedit** [comment]
I tried using cmd method to build windows bundle details below.

Without `<architecture>64</architecture>` this tag getting error below image :
![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/37571867/6e38f953-9411-4104-89cc-c402b4bb260e)

With `<architecture>64</architecture>` this tag got windows build without any error.

My Pc config : 
![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/37571867/b32d436f-cebb-4a39-8d32-4b3b88e02a77)

I had 64 bit windows operating system. Is there any way to build 32bit version windows build from my 64bit operating system.

AIR SDK : 50.2.4.1

@ajwfrost please help ASAP.

**Reply by ajwfrost** [reply]
Hi

That `EndUpdateResource` reference is actually just reflecting a call to the Win32 function:
https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-endupdateresourcew
And the error code 110:
```
ERROR_OPEN_FAILED
110 (0x6E)
The system cannot open the device or file specified.
```

There shouldn't be any problem with a 64-bit version of Windows being used to create the 32-bit binaries here .. most people will have that set-up now. 

Looking at the code, the fact it's throwing from the `EndUpdateResource` method doesn't necessarily mean that's where the error first occurs, it could equally be triggered by an error in `BeginUpdateResource`, and the error maybe implies that there's no file there .. it should be acting on a package/bundle that it's constructing, using the template.exe file and then adjusting the resources within this in order to turn it into your application's executable file. So, can you perhaps check that you have the below file in the SDK:
`...\runtimes\air\win\Adobe AIR\Versions\1.0\Resources\template.exe`
and make sure it's read-write?

In case it's kicking you out halfway through, it may also be worth checking to see if there's any temporary folder still left around that could shed some light on it. Can you go `%TEMP%` in your Windows file explorer, and see if there's a folder created around the time you did the build, name will be in the form of a guid I think. If so, please check what's in there and whether there's an executable file in the root of it...

thanks

**Comment by balaedit** [comment]
Cleared %temp% folder and remaining list of files in that folder below :
![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/37571867/f2d90f04-30b9-4a61-9493-1101a12dc9f0)

Cleared temp folder and remaining list of files in that folder below :
![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/37571867/97f38eb6-5211-4cee-9347-723e39d06587)

First I removed the <architecture>64</architecture> tag from xml description and build failed and then added the tag and build sucessful below is the image:
![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/37571867/10dfbdee-b615-448b-9b60-83dcb8d7a55c)

Is there any other location to clear/delete the Harman cache file. Please suggest.

Thanks @ajwfrost for early reply.

**Comment by balaedit** [comment]
I installed VMware and loaded windows 10 32bit OS and got successfully got windows 32bit bundle of my App without any error.

**Comment by balaedit** [comment]
downloaded the latest Harman AIR SDK 50.2.4.3, same error as above. But get 32 bit bundle of my app in vmware with windows 10 32 bit OS.

When will it resolve so that in my 64 bit OS I can build 32 bit app. @ajwfrost pls is there any other way please suggest.
